core/trino-spi/src/main/java/io/trino/spi/type/SmallintType.java

```diff
@@ -138,7 +138,7 @@ public void appendTo(Block block, int position, BlockBuilder blockBuilder)
     @Override
     public long getLong(Block block, int position)
     {
-        return (long) block.getShort(position, 0);
+        return block.getShort(position, 0);
```

Comment:
Let's revert this change. Other types also have redundant cast now, e.g, `TinyintType`. Changing only `SmallintType` doesn't make sense to me.

Reply:
Good catch, I checked all the implementations of `AbstractType#getLong` and only `TinyintType` and `SmallintType` have this issue, so I think I should fix all the places instead of revert, do you agree?

Reply:
Could you separate into two commits? 
* Remove redundant cast from SmallintType and TinyintType
* Fix grammar and typo in JdbcPageSink